src/errors.rs

```diff
@@ -40,6 +41,10 @@ pub enum Error {
     InvalidEntryActions(BTreeMap<Vec<u8>, EntryError>),
     /// Key does not exist
     NoSuchKey,
+    /// Key already exists
+    KeyExists(Entries),
+    /// Duplicate Entries in this push
+    DuplicateEntries,
```

Comment:
This error is returned if the _keys_ are duplicate.

Reply:
I guess to avoid confusion with crypto keys though, this could maybe be renamed to `DuplicateEntryKeys` rather than just `DuplicateKeys`?